### PR TITLE
[KO] Update style_transfer.ipynb

### PR DESCRIPTION
**Pull request by jaketae**
Partially addresses #184

### CODE REVIEW

File: site/ko/tutorials/generative/style_transfer.ipynb [hunk 1]
```diff
@@ -488,84 +489,67 @@
...
-        "콘텐츠 손실 함수의 정의는 실제로 매우 간단합니다. 먼저 원하는 콘텐츠 이미지와 기본 입력 이미지를 네트워크에 전달합니다. 이렇게하면 모델에서 위에서 정의되었던 중간층들의 출력을 반환할 것입니다. 그 후 우리는 간단히 이 두 이미지의 중간 표현(representation) 사이의 유클리디안 거리(euclidean distance)를 얻을 수 있습니다.\n",
+        "## 스타일 계산하기\n",
         "\n",
-        "좀 더 공식적으로, 콘텐츠 손실(content loss)은 출력이미지 $x$로 부터 콘텐츠 이미지 $p$까지의 콘텐츠의 거리를 표현한 함수입니다. $C_{nn}$을 사전 학습된 딥 합성곱 신경망 네트워크(deep convolutional neural network)라고 합시다. 다시, 이 경우 [VGG19](https://keras.io/applications/#vgg19)모델을 사용합니다. $X$를 어느 이미지라고 하면, $C_{nn}(X)$은 X로 주어진 네트워크입니다. layer $l$에서 입력값이 $x$와 $p$일때 $F^l_{ij}(x) \\in C_{nn}(x)$ 와 $P^l_{ij}(p) \\in C_{nn}(p)$가 중간 특성 표현(feature representation)을 나타낸다고 하면, 콘텐츠 거리(손실)는 다음과 같이 표현할 수 있습니다.\n",
-        "$$L^l_{content}(p, x) = \\sum_{i, j} (F^l_{ij}(x) - P^l_{ij}(p))^2$$\n",
+        "이미지의 콘텐츠는 중간층들의 특성 맵(feature map)의 값들로 표현됩니다.\n",
         "\n",
-        "우리는 일반적인 방식으로 역전파(backpropagation) 과정을 수행하여 콘텐츠 손실(loss)를 최소화 시킬 것입니다. 따라서 특정층(콘텐츠 레이어로 정의 된)에서 원본 이미지와 비슷한 응답을 가질때까지 초기 이미지를 변화시킬것입니다.\n",
+        "이미지의 스타일은 각 피쳐맵의 평균과 피쳐맵들 사이의 상관관계로 설명할 수 있습니다. 이런 정보를 담고 있는 그람 행렬(Gram matrix)은 각 위치에서 특성 벡터(feature vector)끼리의 외적을 구한 후,평균값을 냄으로써 구할 수 있습니다. 주어진 층에 대한 그람 행렬은 다음과 같이 계산할 수 있습니다:\n",
```
Comment (rickiepark) [inline site/ko/tutorials/generative/style_transfer.ipynb:548]: Two '피처맵' --> '특성 맵'

File: site/ko/tutorials/generative/style_transfer.ipynb [hunk 2]
```diff
@@ -574,53 +558,86 @@
...
+        "    super(StyleContentModel, self).__init__()\n",
+        "    self.vgg =  vgg_layers(style_layers + content_layers)\n",
+        "    self.style_layers = style_layers\n",
+        "    self.content_layers = content_layers\n",
+        "    self.num_style_layers = len(style_layers)\n",
+        "    self.vgg.trainable = False\n",
         "\n",
-        "수학적으로, 우리는 기본 입력 이미지 $x$와 스타일 이미지 $a$의 스타일 손실을 이들의 스타일 표현(the gram matrix)사이의 거리로 표현했습니다. 또한 layer $l$에서 벡터화된 특성맵 $i$ and $j$ 사이의 내부 산물을 $G^l_{ij}$로 정의할때, 이미지의 스타일 표현은 그람 행렬 $G^l$에 의해 주어진 필터 응답(filter response) 사이의 상관관계로 표현됩니다. 주어진 이미지의 특성맵 위에 생성된 $G^l_{ij}$은 특성맵 $i$ and $j$ 사이의 상관관계를 나타내는 것을 볼 수 있습니다.\n",
+        "  def call(self, inputs):\n",
+        "    \"[0,1] 사이의 플로트값을 입력으로 받습니다\"\n",
```
Comment (rickiepark) [inline site/ko/tutorials/generative/style_transfer.ipynb:612]: '플로트값을' --> '실수 값을' or 'float 값을'

File: site/ko/tutorials/generative/style_transfer.ipynb [hunk 3]
```diff
@@ -629,61 +646,68 @@
...
-        "  # 손실은 주어진 특성맵의 크기와 필터의 개수에 따라 확장됩니다.\n",
-        "  height, width, channels = base_style.get_shape().as_list()\n",
-        "  gram_style = gram_matrix(base_style)\n",
+        "print('Styles:')\n",
+        "for name, output in sorted(results['style'].items()):\n",
+        "  print(\"  \", name)\n",
+        "  print(\"    shape: \", output.numpy().shape)\n",
+        "  print(\"    min: \", output.numpy().min())\n",
+        "  print(\"    max: \", output.numpy().max())\n",
+        "  print(\"    mean: \", output.numpy().mean())\n",
```
Comment (rickiepark) [inline site/ko/tutorials/generative/style_transfer.ipynb:663]: Please translate Styles, shape, min, max, mean.

File: site/ko/tutorials/generative/style_transfer.ipynb [hunk 4]
```diff
@@ -629,61 +646,68 @@
...
+        "  print()\n",
         "\n",
-        "  return tf.reduce_mean(tf.square(gram_style - gram_target))# / (4. * (channels ** 2) * (width * height) ** 2)"
+        "print(\"Contents:\")\n",
+        "for name, output in sorted(results['content'].items()):\n",
+        "  print(\"  \", name)\n",
+        "  print(\"    shape: \", output.numpy().shape)\n",
+        "  print(\"    min: \", output.numpy().min())\n",
+        "  print(\"    max: \", output.numpy().max())\n",
+        "  print(\"    mean: \", output.numpy().mean())\n"
```
Comment (rickiepark) [inline site/ko/tutorials/generative/style_transfer.ipynb:672]: Please translate Contents, shape, min, max, mean.

File: site/ko/tutorials/generative/style_transfer.ipynb [hunk 5]
```diff
@@ -629,61 +646,68 @@
...
-        "id": "pXIUX6czZABh"
+        "id": "y9r8Lyjb_m0u"
       },
       "source": [
-        "## 이미지에 스타일 변형 적용하기"
+        "## 경사하강법 실행\n",
+        "\n",
+        "이제 스타일과 콘텐츠 추출기를 사용해 스타일 전이 알고리즘을 구현할 차례입니다. 타겟에 대한 입력 이미지의 평균 제곱 오차를 계산한 후, 오차값들의 가중합을 구합니다.\n",
+        "\n",
+        "스타일과 콘텐츠의 타겟값을 지정합니다:"
```
Comment (rickiepark) [inline site/ko/tutorials/generative/style_transfer.ipynb:686]: Two '타겟'-->'타깃'

File: site/ko/tutorials/generative/style_transfer.ipynb [hunk 6]
```diff
@@ -629,61 +646,68 @@
...
     {
       "cell_type": "markdown",
       "metadata": {
         "colab_type": "text",
-        "id": "-kGzV6LTp4CU"
+        "id": "CNPrpl-e_w9A"
       },
       "source": [
-        "콘텐츠와 스타일 이미지를 적재하고, 네트워크에 피드를 제공하며, 모델에서 콘텐츠와 스타일 특성 표현을 출력하는 헬퍼 함수(helper function)을 정의합니다."
+        "최적화 시킬 이미지를 담을 `tf.Variable`을 정의하고 콘텐츠 이미지로 초기화합니다. (이때 `tf.Variable`는 콘텐츠 이미지와 크기가 같아야 합니다.):"
```
Comment (rickiepark) [inline site/ko/tutorials/generative/style_transfer.ipynb:710]: '최적화 시킬' --> '최적화시킬'

File: site/ko/tutorials/generative/style_transfer.ipynb [hunk 7]
```diff
@@ -692,49 +716,21 @@
...
       "cell_type": "markdown",
       "metadata": {
         "colab_type": "text",
-        "id": "3DopXw7-lFHa"
+        "id": "M6L8ojmn_6rH"
       },
       "source": [
-        "### 손실과 그래디언트 계산하기\n",
-        "그래디언트를 계산하기위해서 [**tf.GradientTape**](https://www.tensorflow.org/programmers_guide/eager#computing_gradients)를 사용하였습니다. 이는 나중에 그래디언트 계산 과정을 추적하여 사용할 수 있는 automatic differentiation 기능을 활용할 수 있습니다. 이것은 정방향 전파동안의 명령들을 기록하여 역전파시 입력 이미지와 관련하여 손실 함수의 그래디언트를 계산할 수 있게 합니다."
+        "픽셀값을 0과 1 사이의 값으로 정규화해주는 함수를 정의합니다:"
```
Comment (rickiepark) [inline site/ko/tutorials/generative/style_transfer.ipynb:733]: Readers may confuse '정규화' with 'normalization'.
plt.imshow draw pixels with 0-1 float or 0-255 int.
So I suggest,
'픽셀값을 0과 1 사이의 값으로 정규화해주는 함수를 정의합니다' --> '픽셀 값이 실수이므로 0과 1 사이로 클리핑하는 함수를 정의합니다'

File: site/ko/tutorials/generative/style_transfer.ipynb [hunk 8]
```diff
@@ -1001,27 +891,40 @@
...
+        "  for m in range(steps_per_epoch):\n",
+        "    step += 1\n",
+        "    train_step(image)\n",
+        "    print(\".\", end='')\n",
+        "  display.clear_output(wait=True)\n",
+        "  display.display(tensor_to_image(image))\n",
+        "  print(\"Train step: {}\".format(step))\n",
+        "  \n",
+        "end = time.time()\n",
+        "print(\"Total time: {:.1f}\".format(end-start))"
```
Comment (rickiepark) [inline site/ko/tutorials/generative/style_transfer.ipynb:915]: Please translate 'Train steps' and 'Total time'.

File: site/ko/tutorials/generative/style_transfer.ipynb [hunk 9]
```diff
@@ -1001,27 +891,40 @@
...
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "GWVB3anJMY2v"
+      },
+      "source": [
+        "## 전체 분산 오차\n",
```
Comment (rickiepark) [inline site/ko/tutorials/generative/style_transfer.ipynb:925]: Readers may confuse '분산' with variance.
I suggest,
'전체 분산 오차' --> '총 변위 손실'

File: site/ko/tutorials/generative/style_transfer.ipynb [hunk 10]
```diff
@@ -1001,27 +891,40 @@
...
         "\n",
-        "    plt.imshow(best_img)\n",
-        "    plt.title('Output Image')\n",
-        "    plt.show()"
+        "이 기본 구현 방식의 한 가지 단점은 많은 고주파 아티팩(high frequency artifact)가 생겨난다는 점 입니다. 아티팩 생성을 줄이기 위해서는 이미지의 고주파 구성 요소에 대한 정규화(regularization) 항을 추가해야 합니다. 스타일 전이에서는 이 변형된 오차값을 *전체 분산 오차(total variation loss)*라고 합니다:"
```
Comment (rickiepark) [inline site/ko/tutorials/generative/style_transfer.ipynb:927]: How about '정규화' --> '규제'?
Because we also translate normalization as '정규화'. :)

File: site/ko/tutorials/generative/style_transfer.ipynb [hunk 11]
```diff
@@ -1105,12 +1034,21 @@
...
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "pu2hJ8zOKMc1"
+      },
+      "source": [
+        "식이 잘 게산된다는 것을 확인할 수 있습니다. 하지만 다행히도 텐서플로에는 이미 표준 함수가 내장되어 있기 직접 오차식을 구현할 필요는 없습니다:"
```
Comment (rickiepark) [inline site/ko/tutorials/generative/style_transfer.ipynb:1051]: '게산' --> '계산'

File: site/ko/tutorials/generative/style_transfer.ipynb [hunk 12]
```diff
@@ -1197,47 +1160,59 @@
...
+        "  for m in range(steps_per_epoch):\n",
+        "    step += 1\n",
+        "    train_step(image)\n",
+        "    print(\".\", end='')\n",
+        "  display.clear_output(wait=True)\n",
+        "  display.display(tensor_to_image(image))\n",
+        "  print(\"Train step: {}\".format(step))\n",
+        "\n",
+        "end = time.time()\n",
+        "print(\"Total time: {:.1f}\".format(end-start))"
```
Comment (rickiepark) [inline site/ko/tutorials/generative/style_transfer.ipynb:1184]: Please translate 'Train step' and 'Total time'.

File: site/ko/tutorials/generative/style_transfer.ipynb (hunk 10 above)
Comment (jaketae) [reply site/ko/tutorials/generative/style_transfer.ipynb:927]: I decided to change it to '레귤러리제이션' instead. Does this sound okay to you?

File: site/ko/tutorials/generative/style_transfer.ipynb (hunk 8 above)
Comment (jaketae) [reply site/ko/tutorials/generative/style_transfer.ipynb:915]: Does '훈련 스텝' sound like a natural translation of 'Train steps'?

**Comment by jaketae** [comment]
@rickiepark Thank you so much for the detailed input! I've reflected most of the changes you've suggested, with the exception of the two questions I have posted as comments. Thanks again!

File: site/ko/tutorials/generative/style_transfer.ipynb (hunk 8 above)
Comment (rickiepark) [reply site/ko/tutorials/generative/style_transfer.ipynb]: '훈련 스텝' is good, because '단계' is used too much. :)

File: site/ko/tutorials/generative/style_transfer.ipynb (hunk 10 above)
Comment (rickiepark) [reply site/ko/tutorials/generative/style_transfer.ipynb]: '레귤러리제이션' is good. :)

**Comment by rickiepark** [comment]
Thank you so much @jaketae 
LGTM 👍
